### PR TITLE
Wrong/broken Changelog: Add lobby size widget and player join/leave notifications

### PR DESCRIPTION
**Pull request by brandonwamboldt**

### CODE REVIEW

File: src/main/java/at/hannibal2/skyhanni/features/misc/SocialFeatures.kt [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    @SubscribeEvent
+    fun onRenderOverlay(event: GuiRenderEvent.GuiOverlayRenderEvent) {
+        if (!LorenzUtils.inSkyBlock || LorenzUtils.inDungeons || !config.showLobbySize) return;
+        config.showLobbySizePos?.renderString("${players.size} Players", posLabel = "Players in Lobby")
+    }
+
+    @SubscribeEvent
+    fun onWorldChange(event: LorenzWorldChangeEvent) {
+        players = mutableListOf<String>()
```
Comment (Thunderblade73) [inline src/main/java/at/hannibal2/skyhanni/features/misc/SocialFeatures.kt:25]: use ```players.clear()``` and change players to a val

File: src/main/java/at/hannibal2/skyhanni/features/misc/SocialFeatures.kt [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    @SubscribeEvent
+    fun onTabListUpdate(event: TabListUpdateEvent) {
+        if (!LorenzUtils.inSkyBlock || LorenzUtils.inDungeons) return;
+        if (!config.playerLeaveNotifications && !config.playerJoinNotifications) return;
+
+        val playersInfo = TabListData.getTabList().filter { it.startsWith("§8[" ) }.map { it.removeColor().split("] ").last().split(" ").first() }.distinct()
+        val initialLoad = players.isEmpty()
```
Comment (Thunderblade73) [inline src/main/java/at/hannibal2/skyhanni/features/misc/SocialFeatures.kt:34]: change to ```if(players.isEmpty()) return``` no need for the variable

File: src/main/java/at/hannibal2/skyhanni/features/misc/SocialFeatures.kt [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    fun onWorldChange(event: LorenzWorldChangeEvent) {
+        players = mutableListOf<String>()
+    }
+
+    @SubscribeEvent
+    fun onTabListUpdate(event: TabListUpdateEvent) {
+        if (!LorenzUtils.inSkyBlock || LorenzUtils.inDungeons) return;
+        if (!config.playerLeaveNotifications && !config.playerJoinNotifications) return;
+
+        val playersInfo = TabListData.getTabList().filter { it.startsWith("§8[" ) }.map { it.removeColor().split("] ").last().split(" ").first() }.distinct()
```
Comment (Thunderblade73) [inline src/main/java/at/hannibal2/skyhanni/features/misc/SocialFeatures.kt:33]: change to regex. 
Somewhat like this (not tested):
```Kotlin
val regex = "\\].{5}(\\S)+".toRegex()
TabListData.getTabList().mapNotNull{regex.find(it)?.groupValues?.get(1)}
```

File: src/main/java/at/hannibal2/skyhanni/features/misc/SocialFeatures.kt [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        }
+
+        for (player in players) {
+            if (!playersInfo.contains(player) && config.playerLeaveNotifications && config.playerNotificationLobbySize >=  playersInfo.size) {
+                LorenzUtils.chat("§7${player} has left your lobby", false)
+            }
+        }
+
+        players = playersInfo.toMutableList()
+    }
```
Comment (Thunderblade73) [inline src/main/java/at/hannibal2/skyhanni/features/misc/SocialFeatures.kt:51]: make players and playerInfo a ```set``` then the code becomes much simpler
```Kotlin
val leave = players - playerInfo
if(config.playerLeaveNotifications && config.playerNotificationLobbySize >= playersInfo.size){
  leave.foreach{
    LorenzUtils.chat("§7${player} has left your lobby", false)
  }
}
leave.foreach{
  players.remove(it)
}
val join = playerInfo - player
if(config.playerJoinNotifications && config.playerNotificationLobbySize >= playersInfo.size){
  join.foreach{
    LorenzUtils.chat("§7$it has joined your lobby", false)
  }
}
join.foreach{
  players.add(it)
}
```

**Review by Thunderblade73 — CHANGES_REQUESTED** [review]

**Review by hannibal002 — CHANGES_REQUESTED** [review]
Thunder's Changes

**Pull request closed without merging**